### ISSUE TITLE
`fee too low` error on Truffle migration or test when using ops tool

### ISSUE THREAD

**Issue by platocrat** [ethereum-optimism/optimism#1081]
**Describe the bug**
Users in Discord reported errors when using Kovan **AND** the ops tool.
Running Truffle tests results in a `fee too low error`.
```
Error: while migrating ERC20: fee too low: 100829625000000, 
use at least tx.gasLimit = 54181975 and tx.gasPrice = 15000000
```

Note that this error is _after_ hardcoding a `gasLimit` and `gasPrice` that matches the error log's suggestion (also, based on user reports, I had different suggested values than those reported).

**To Reproduce**
1. `git clone https://github.com/ethereum-optimism/optimism.git`
2. Pull recent changes from develop branch: `git pull origin develop`
3. Start docker services: `yarn && yarn build && cd ops && docker-compose build && docker-compose up`
4. In a new terminal, run Truffle tests under `optimism/examples/truffle`:  `yarn truffle test --network optimism --config truffle-config-ovm.js`
5. See error:
```
Error: while migrating ERC20: fee too low: 100829625000000, 
use at least tx.gasLimit = 54181975 and tx.gasPrice = 15000000
```

**Expected behavior**
Tests should pass after adding the suggested gasLimit and gasPrice values

**System Specs:**
 - OS: MacOS
 - [Latest commit of `develop` branch](https://github.com/ethereum-optimism/optimism/commit/6d7b12b8693117c379be76223448ef36707304d2)

**Comment by JarenL** [comment]
Similar issue.

Using Truffle v5.3.9 

`? Pick a network optimistic_kovan
✖ Preparing everything to publish the project. Deploying new App contract
fee too low: 120000000000000, use at least tx.gasLimit = 80430000 and tx.gasPrice = 15000000`

**Comment by tynes** [comment]
What is your `tx.gasLimit` and `tx.gasPrice`? I believe truffle hard codes the gas price for deployment txs

**Comment by rube-de** [comment]
any update on this? running the truffle-erc20-example and being unable to follow it because of this error.

**Comment by JarenL** [comment]
@tynes I used Optimism suggested then tried with the values from the error, no change error still persists.

**Comment by djomaa** [comment]
It seems truffle estimates too low gas limit. 
For me hardcoding `gasLimit` from the error to `deployer.deploy(Contract, { gasLimit: <value from error> })` and setting `gasPrice` in truffle-config worker to `15000000`.

**Comment by qbzzt** [comment]
This error is still happening with the latest code. It might be a clue that 

```javascript
x = await SimpleStorage.new()
```

Is successful, but 

```javascript
y = await Counter.new()
```

fails with the error. I'll attach the two contracts here - they are both tiny.

[Counter.sol.txt](https://github.com/ethereum-optimism/optimism/files/6833969/Counter.sol.txt)
[SimpleStorage.sol.txt](https://github.com/ethereum-optimism/optimism/files/6833970/SimpleStorage.sol.txt)

**Comment by JarenL** [comment]
Is there anyone out there who has deployed on optimism with truffle?

**Comment by tynes** [comment]
To get the example to work, we had to hard code the gas limit in the deployment transaction. Truffle seems to use a hardcoded, constant amount of gas for all deployment transactions, see: https://github.com/ethereum-optimism/optimism/blob/c7bc85deee999b8edfbe187b302d0ea262638ca9/examples/truffle/truffle-config-ovm.js#L20

**Comment by JarenL** [comment]
I tried what you did, direct copy, still same error. I am trying to deploy to optimism-kovan, does that require something different?

**Comment by JarenL** [comment]
Truffle put out a [optimism-box](https://github.com/truffle-box/optimism-box). Followed the steps there for optimism-kovan setup, still running into issues with the whole "fee too low" issue everyone is having.

**Comment by haydenyoung** [comment]
Also running into the fee to low error with the truffle example project https://github.com/ethereum-optimism/Truffle-ERC20-Example. Got the truffle box example working though https://github.com/ethereum-optimism/optimism-tutorial/tree/main/truffle but it is a lot smaller than the first example.

The example results in a gas limit that far exceeds optimisms max tx gaslimit and hence the gaslimit error (setting `gas: 54180130` in the truffle config doesn't work for me either).

A couple of interesting points:

Deploying the contract to ganache yielded a much smaller estimated gas:

ganache: 649757
optimism-ethereum: 54180130

Also, changing the TARGET_GAS_LIMIT in optimism to something much higher did result in a different error (something about insufficient funds).

It seems that the truffle line 

```
token = await ERC20.new(10000, tokenName, tokenDecimals, tokenSymbol, { from: accounts[0] })
```
causes issues on Optimism resulting in a much larger gas limit which is outside the max gas limit of Optimism. My understanding is the gas limit will be higher when deploying contracts on Optimism but is it only 25% because this seems 10x higher.

**Comment by djomaa** [comment]
Estimation gas limit by myself worked for me
[code link](https://rentry.co/74v3a)

**Comment by haydenyoung** [comment]
strangely enough, setting gasPrice: 0 is working for me
```
{ gasPrice: 0 }
````

**Comment by platocrat** [comment]
Is this still an issue?

I noticed that I need to hardcode a gasLimit for all of my transactions that change state.

Also, 
> strangely enough, setting gasPrice: 0 on the tx works for me
> 
> ```
> { gasPrice: 0 }
> ```

Setting `gasPrice` to 0 would not work on Kovan for sending transactions, just a friendly FYI 😃

**Comment by JarenL** [comment]
@platocrat Yes still an issue for me. Cannot deploy anything with truffle and Jovan.
